api/lib/domain/models/Answer.js

```diff
@@ -57,6 +59,10 @@ class Answer {
   get hasTimedOut() {
     return _.isInteger(this.timeout) && this.timeout < 0;
   }
+
+  setTimeSpentFrom({ lastQuestionDate }) {
+    this.timeSpent = Math.ceil((new Date().getTime() - lastQuestionDate.getTime()) / 1000);
```

Comment:
Attention, `new Date` est une dépendance externe qui est référencée directement ici. Certes, `sinon.useFakeTimers` permet de tricher pour les tests mais ça reste dangereux de dépendre de la date système au fond d'une classe métier qui ne déclare même pas cette dépendance.
Il vaudrait mieux traiter la récupération de la date en cours comme une opération d'appel à un système externe. En l'occurrence ici, il suffirait que `Answer.setTimeSpentFrom` prenne en argument la date "en cours", qui devra donc être fournie par le `usecase`, qui lui peut récupérer cette date en déclarant une dépendance à un service chargé de la fournir. Les tests pourraient donc notamment être écrits sans `sinon.useFakeTimers` et la notion de la date en cours serait centralisée, ce qui peut avoir d'autres avantages aussi.
Un des nombreux articles qu'on peut trouver sur ce sujet : http://nf2p.com/dot-net/datetime-now-is-evil/

Reply:
L'appel à `new Date()` est sorti dans [ce commit](https://github.com/1024pix/pix/pull/2705/commits/a9f0d6b8fdf8d879b078709704f4dd9d71c415e4).